runtime/lua/nvim/health.lua

```diff
@@ -23,9 +23,15 @@ local function check_runtime()
   health.start('Runtime')
   -- Files from an old installation.
   local bad_files = {
+    ['plugin/health.vim'] = false,
+    ['autoload/health/nvim.vim'] = false,
+    ['autoload/health/provider.vim'] = false,
+    ['autoload/man.vim'] = false,
     ['plugin/man.vim'] = false,
+    ['queries/help/highlights.scm'] = false,
+    ['queries/help/injections.scm'] = false,
```

Comment:
Added some more checks for old runtime files. 

- The old provider files can cause `:checkhealth` errors.
- `queries/help/` is mostly defensive. @clason